app/controllers/processed_applications_controller.rb

```diff
@@ -1,31 +1,29 @@
 class ProcessedApplicationsController < ApplicationController
-  before_action :authenticate_user!
-
   include ProcessedViewsHelper
   helper ReferenceHelper
 
   def index
-    @applications = Query::ProcessedApplications.new(current_user).find.map do |application|
+    authorize :application
+
+    @applications = applications.map do |application|
       Views::ApplicationList.new(application)
     end
   end
 
   def show
+    authorize application
+
     @form = Forms::Application::Delete.new(application)
     assign_views
   end
 
   def update
+    # TODO: The authorisation should be done after the attributes have been updated
```

Comment:
Yep, it is. It's related to the fact that we're updating the `Application` inside the form object and therefore we can't easily do authorisation after the attributes have been assigned, but before saved.